### PR TITLE
Feature/crafttweaker integration

### PR DESCRIPTION
**Pull request by jaredlll08**
This supersedes #4563 and targets the 1.16.5 branch.

This is marked as a draft while we discuss @SkySom's comment on $4563
> Would it make sense to have count somehow be applied within CraftTweaker to IIngredients? I also use count on Tags and such in Transport, and definitely see it as a useful addition (And I think it is something that is useful? Only having ItemStacks be able to have count for the integration with IE seems slightly less useful?)

### CODE REVIEW

**Review by SkySom — COMMENTED** [review]
Looks good to me. Will pop to approval once the count stuff is fixed it.

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/crafttweaker/CrTIngredientUtil.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	 * So far, only IItemStack supports setting an amount.
+	 * So we can at least check for that, I guess?
+	 */
+	public static IngredientWithSize getIngredientWithSize(IIngredient crafttweakerIngredient)
+	{
+		final Ingredient basePredicate = crafttweakerIngredient.asVanillaIngredient();
+		if(crafttweakerIngredient instanceof IItemStack)
+		{
+			return IngredientWithSize.of(((IItemStack)crafttweakerIngredient).getInternal());
+		}
```
Comment (malte0811) [inline src/main/java/blusunrize/immersiveengineering/common/util/compat/crafttweaker/CrTIngredientUtil.java]: Formatting (braces should be omitted for single-statement blocks). This is consistent in the entire diff, so I won't comment on other cases.

**Review by malte0811 — COMMENTED** [review]
Other than the formatting issue the PR looks good to me.

**Review by BluSunrize — APPROVED** [review]
Ideally the copyright notices should say 2021 on them, given that that's the their commit date as well. But beyond that, LGTM

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/crafttweaker/CrTIngredientUtil.java (hunk 1 above)
Comment (jaredlll08) [reply src/main/java/blusunrize/immersiveengineering/common/util/compat/crafttweaker/CrTIngredientUtil.java]: Applied the codestyle, so this should all be good now ^^

**Comment by jaredlll08** [comment]
Update (and added where it was missing) the license header, this should be good to go!

**Review by SkySom — APPROVED** [review]

**Pull request merged**